traitsui/testing/tester/examples/range_example.py

```diff
@@ -0,0 +1,54 @@
+""" Example for utilizing UI Tester on various modes of Range editor.
+
+In this example, the Tester locates the textbox for the each style of Range
+Editor and modifies the value.
+"""
+import platform
+
+from traits.api import HasTraits, Int
+from traitsui.api import Item, RangeEditor, View
+from traitsui.testing.tester import command, locator, query
+from traitsui.testing.tester.ui_tester import UITester
+from traitsui.tests._tools import is_wx
+is_windows = platform.system() == "Windows"
+
+
+class RangeModel(HasTraits):
+    """ Dummy class holding the Int trait to be displayed with a ListEditor.
+    """
+
+    value = Int(1)
+
+
+def get_view(mode):
+    """ Gets a view with a specific mode """
+    return View(
+        Item(
+            "value",
+            editor=RangeEditor(low=1, high=12, mode=mode)
+        )
+    )
+
+
+model = RangeModel()
+
+if __name__ == '__main__':
+    tester = UITester(delay=500)
+    for mode in ['slider', 'xslider', 'logslider', 'text']:
+        with tester.create_ui(model, dict(view=get_view(mode))) as ui:
+            number_field = tester.find_by_name(ui, "value")
+            text = number_field.locate(locator.WidgetType.textbox)
+
+            if is_windows and is_wx() and mode == 'text':
+                # For RangeTextEditor on wx and windows, the textbox
+                # automatically gets focus and the full content is selected.
+                # Insertion point is moved to keep the test consistent
+                text.target.textbox.SetInsertionPointEnd()
```

Comment:
If we are exposing this test as a showcase example, I'd very much like to implement `KeyClick("End")` for wx TextEntry.